homeassistant/components/automation/__init__.py

```diff
@@ -305,101 +304,68 @@ def cond_func(variables):
                     """Condition will always pass."""
                     return True
 
-            attach_triggers = partial(_process_trigger, hass, config,
-                                      config_block.get(CONF_TRIGGER, []), name)
-            entity = AutomationEntity(name, attach_triggers, cond_func, action,
-                                      hidden)
-            component.add_entities((entity,))
+            async_attach_triggers = partial(
+                _async_process_trigger, hass, config,
+                config_block.get(CONF_TRIGGER, []), name)
+            entity = AutomationEntity(name, async_attach_triggers, cond_func,
+                                      action, hidden)
+            yield from hass.loop.run_in_executor(
```

Comment:
I initially explored that idea but that let me into a big rabbit hole. The entity component interfaces with a lot of sync things that are not ready yet to be all made async. It will happen but I didn't do it this time as this is already getting pretty big.

However, I will move the executor call to add all entities at once to address your concern 👍